linux_os/guide/services/ssh/ssh_client/ssh_use_approved_macs_ordered_stig/oval/shared.xml

```diff
@@ -0,0 +1,65 @@
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("Limit the Message Authentication Codes (MACs) to those which are FIPS-approved.", rule_title=rule_title) }}}
+    <criteria comment="ssh is configured correctly" operator="AND">
+      <criteria comment="the configuration is correct if it exists" operator="AND">
+        <criterion comment="Check the MACs in /etc/ssh/ssh_config if any"
+          test_ref="test_{{{ rule_id }}}" />
+        <criterion comment="Check the MACs in /etc/ssh/ssh_config.d if any"
+          test_ref="test_{{{ rule_id }}}_config_dir" />
+      </criteria>
+      <criterion comment="the configuraton exists" test_ref="test_MACs_present_{{{ rule_id }}}" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="any_exist"
+    comment="tests the value of MACs setting in the /etc/ssh/ssh_config file"
+    id="test_{{{ rule_id }}}" version="1">
+    <ind:object object_ref="obj_{{{ rule_id }}}" />
+    <ind:state state_ref="state_{{{ rule_id }}}" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="obj_{{{ rule_id }}}" version="1">
+    <ind:filepath>/etc/ssh/ssh_config</ind:filepath>
+    <ind:pattern operation="pattern match">^[ \t]*(?i)MACs(?-i)[ \t]+(.+?)[ \t]*(?:$|#)</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test check="all" check_existence="any_exist"
+    comment="tests the value of MACs setting in the /etc/ssh/ssh_config.d file"
+    id="test_{{{ rule_id }}}_config_dir" version="1">
+    <ind:object object_ref="obj_{{{ rule_id }}}_config_dir" />
+    <ind:state state_ref="state_{{{ rule_id }}}_config_dir" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="obj_{{{ rule_id }}}_config_dir" version="1">
+    <ind:path>/etc/ssh/ssh_config.d</ind:path>
+    <ind:filename operation="pattern match">.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^[ \t]*(?i)MACs(?-i)[ \t]+(.+?)[ \t]*(?:$|#)</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_state id="state_{{{ rule_id }}}" version="1">
+    <ind:subexpression operation="equals" datatype="string" var_ref="ssh_approved_macs" var_check="at least one" />
+  </ind:textfilecontent54_state>
+
+  <ind:textfilecontent54_state id="state_{{{ rule_id }}}_config_dir" version="1">
+    <ind:subexpression operation="equals" datatype="string" var_ref="ssh_approved_macs" var_check="at least one" />
+  </ind:textfilecontent54_state>
```

Comment:
the 2 state elements are effectively the same, there could be only one

Reply:
Fixed